communication/mavlink_waypoint_handler.cpp

```diff
@@ -288,7 +288,6 @@ void Mavlink_waypoint_handler::receive_waypoint(Mavlink_waypoint_handler* waypoi
                             print_util_dbg_print("flight plan received!\n");
                             waypoint_handler->is_receiving_waypoint_ = false;
 
-                            waypoint_handler->navigation_.set_start_wpt_time();
```

Comment:
I think waypoint_handler should have a method to get the last time the waypoint list was received. 
Mission_planner could check this time, and decide to change the current goal if necessary ( `waypoint_handler_.last_update() > navigation_.start_wpt_time()`)